### PR TITLE
Keystone integration

### PR DESCRIPTION
**Pull request by tyb0807**
Attempt to tackle #843 
@rhelmot please take a look. Thank you

### CODE REVIEW

File: angr/manager.py [hunk 1]
```diff
@@ -585,8 +585,9 @@ def step(self, n=None, selector_func=None, step_func=None, stash=None,
         :param backup_state:    A state to read bytes from instead of using project memory.
         :param opt_level:       The VEX optimization level to use.
         :param insn_bytes:      A string of bytes to use for the block instead of the project.
+        :param insn_text:       A string of assembly instruction to use for the block instead of the project.
         :param size:            The maximum size of the block, in bytes.
-        :param num_inst:        The maximum number of instructions.
+        :param tum_inst:        The maximum number of instructions.
```
Comment (ltfish) [inline angr/manager.py]: Why changing `num` to `tum`?

**Comment by ltfish** [comment]
This title makes me so happpppy!

I guess you want to modify `setup.py` and `requirements.txt` to make sure keystone is installed when installing angr.

**Comment by rhelmot** [comment]
The keystone dependency should be added here, not archinfo.

File: tests/test_keystone.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    'i386': [ 0x080485C9 ],
+    'x86_64': [ 0x4006ed ],
+    'armel': [ 0x85F0 ],
+}
+
+def run_keystone(arch):
+    p = angr.Project(os.path.join(test_location, arch, "fauxware"))
+
+    block = p.factory.block(target_addrs[arch][0])
+    sm = p.factory.simgr()
```
Comment (rhelmot) [inline tests/test_keystone.py]: I think there was a consensus among the dev team that `factory.simgr` isn't super understandable, so we're using the longform `factory.simulation_manager` now

File: tests/test_keystone.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    for i in block.capstone.insns:
+        insn_text += i.mnemonic + ' ' + i.op_str + ';'
+
+    new_block = p.factory.block(block.addr, insn_text=insn_text[:-1], size=block.size)
+
+    nose.tools.assert_equal(block.bytes, new_block.bytes)
+
+    sm.step(addr=block.addr, insn_text=insn_text)
+
+    nose.tools.assert_true('active' in sm.stashes)
```
Comment (rhelmot) [inline tests/test_keystone.py]: Can there be a more interesting assertion here, verifying that the text was actually what got executed?

**Comment by tyb0807** [comment]
@ltfish @rhelmot tests pass now. But I don't understand the colorguard fail though...

**Comment by rhelmot** [comment]
One of the color guard failures is definitely because I bumped the version we’re using to 5.10. The other one might be the same but it’s hard to tell.

**Comment by rhelmot** [comment]
Thanks! This is a great feature.

**Pull request merged**